airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/SentryJobErrorReportingClient.java

```diff
@@ -0,0 +1,123 @@
+
+  static final String STACKTRACE_PARSE_ERROR_TAG_KEY = "stacktrace_parse_error";
+  private final IHub sentryHub;
+  private final SentryExceptionHelper exceptionHelper;
+
+  SentryJobErrorReportingClient(final IHub sentryHub, final SentryExceptionHelper exceptionHelper) {
+    this.sentryHub = sentryHub;
+    this.exceptionHelper = exceptionHelper;
+  }
+
+  public SentryJobErrorReportingClient(final String sentryDSN, final SentryExceptionHelper exceptionHelper) {
+    this(createSentryHubWithDSN(sentryDSN), exceptionHelper);
+  }
+
+  static IHub createSentryHubWithDSN(final String sentryDSN) {
+    if (sentryDSN == null || sentryDSN.isEmpty()) {
+      return NoOpHub.getInstance();
+    }
+
+    final SentryOptions options = new SentryOptions();
+    options.setDsn(sentryDSN);
+    options.setAttachStacktrace(false);
+    options.setEnableUncaughtExceptionHandler(false);
+    return new Hub(options);
+  }
+
+  /**
+   * Reports a Connector Job FailureReason to Sentry
+   *
+   * @param workspace - Workspace where this failure occurred
+   * @param failureReason - FailureReason to report
+   * @param dockerImage - Tagged docker image that represents the release where this failure occurred
+   * @param metadata - Extra metadata to set as tags on the event
+   */
+  @Override
+  public void reportJobFailureReason(final StandardWorkspace workspace,
+                                     final FailureReason failureReason,
+                                     final String dockerImage,
+                                     final Map<String, String> metadata) {
+    final SentryEvent event = new SentryEvent();
+
+    // Remove invalid characters from the release name, use @ so sentry knows how to grab the tag
+    // e.g. airbyte/source-xyz:1.2.0 -> source-xyz@1.2.0
+    // More info at https://docs.sentry.io/product/cli/releases/#creating-releases
+    final String release = dockerImage.replace(":", "@").substring(dockerImage.lastIndexOf("/") + 1);
+    event.setRelease(release);
+
+    // enhance event fingerprint to ensure separate grouping per connector
+    final String[] releaseParts = release.split("@");
+    if (releaseParts.length > 0) {
+      event.setFingerprints(List.of("{{ default }}", releaseParts[0]));
+    }
+
+    // set workspace as the user in sentry to get impact and priority
+    final User sentryUser = new User();
+    sentryUser.setId(String.valueOf(workspace.getWorkspaceId()));
+    sentryUser.setUsername(workspace.getName());
+    event.setUser(sentryUser);
+
+    // set metadata as tags
+    event.setTags(metadata);
+
+    // set failure reason's internalMessage as event message
+    // Sentry will use this to fuzzy-group if no stacktrace information is available
+    final Message message = new Message();
+    message.setFormatted(failureReason.getInternalMessage());
+    event.setMessage(message);
+
+    // events can come from any platform
+    event.setPlatform("other");
+
+    // attach failure reason stack trace
```

Comment:
leading on from my earlier comment, could we have logic here for "if stacktrace couldn't be parsed, send it as a string"?
The matching we saw in Sentry for this wasn't _terrible_, and I would make an assumption better than internal message alone.